massa-models/src/serialization.rs

```diff
@@ -87,6 +98,72 @@ impl DeserializeVarInt for u32 {
     }
 }
 
+// HERE
+impl DeserializeVarIntV2 for u32 {
+    fn from_varint_bytes_v2(
+        buffer: &[u8],
+        min: Option<Self>,
+        max: Option<Self>,
+    ) -> IResult<&[u8], Self> {
+        let error = Err(nom::Err::Failure(nom::error::Error::new(
+            buffer,
+            nom::error::ErrorKind::Fail,
+        )));
+        match unsigned_varint::decode::u32(buffer) {
```

Comment:
I think you can simplify and bring clarity by writing : 
```rust
let min = min.unwrap_or(u32::MIN);
let min = max.unwrap_or(u32::MAX);

match unsigned_varint::decode::u32(buffer) {
  Ok((v, rest)) if v >= min && v < max => Ok((rest, v))
  _ => error
}
```

I don't know if it change something in optimisation time/CPU/memory etc but it gain clarity.

Also I don't understand why `min` is Included and `max` excluded. I think both should be the same and if you want to bring some flexibility I think you could use something like : https://doc.rust-lang.org/stable/std/ops/enum.Bound.html 

Same for all above.

Reply:
I feel like min included and max excluded is the most common behaviour that's why I went for that, but maybe its better to have something different.

Also I believe my implementation is faster, too many operation are made for a `None` value with yours.

Reply:
Yes it's not common for me that the max is excluded.

Yes I think it's faster but I don't know if the gain here worth the reading. It's a choice. Not a problem for me to keep as you have made.